### PR TITLE
Updated MVC projects to utilize Common repo classes.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Had to add extra dependencies to work around aspnet/XRE#1237.
- Updated src and test projects.

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/ActionConstraints/DefaultActionConstraintProvider.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using Microsoft.Framework.Internal;
 using Microsoft.Framework.DependencyInjection;
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Core/ActionConstraints/DefaultActionConstraintProvider.cs]: sort `using`s

File: src/Microsoft.AspNet.Mvc.Core/project.json [hunk 2]
```diff
@@ -10,6 +10,10 @@
     "Microsoft.AspNet.Hosting": "1.0.0-*",
     "Microsoft.AspNet.Http.Extensions": "1.0.0-*",
     "Microsoft.AspNet.Mvc.Common": { "version": "6.0.0-*", "type": "build" },
+    "Microsoft.Framework.NotNullAttribute.Internal": { "version": "1.0.0-*", "type": "build" },
+    "Microsoft.Framework.PropertyActivator.Internal": { "version": "1.0.0-*", "type": "build" },
+    "Microsoft.Framework.PropertyHelper.Internal": { "version": "1.0.0-*", "type": "build" },
+    "Microsoft.Framework.CopyOnWriteDictionary.Internal": { "version": "1.0.0-*", "type": "build" },
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Core/project.json]: sort

File: src/Microsoft.AspNet.Mvc.ModelBinding/project.json [hunk 3]
```diff
@@ -8,6 +8,9 @@
     "Microsoft.AspNet.Http.Core": "1.0.0-*",
     "Microsoft.AspNet.Http.Extensions": "1.0.0-*",
     "Microsoft.AspNet.Mvc.Common": { "version": "6.0.0-*", "type": "build" },
+    "Microsoft.Framework.NotNullAttribute.Internal": { "version": "1.0.0-*", "type": "build" },
+    "Microsoft.Framework.CopyOnWriteDictionary.Internal": { "version": "1.0.0-*", "type": "build" },
+    "Microsoft.Framework.PropertyHelper.Internal": { "version": "1.0.0-*", "type": "build" },
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.ModelBinding/project.json]: sort

**Comment by dougbu** [comment]
can't see past src/Microsoft.AspNet.Mvc.Razor.Host on GitHub.  could you break this up?  if not I'll checkout the branch locally over the weekend.

**Comment by NTaylorMullen** [comment]
Updated, broke into 3 commits. Feel free to comment on commits individually (even though we normally don't do that). That should solve the length issues.

**Comment by dougbu** [comment]
:shipit: after addressing small-ish comments

**Pull request merged**